src/table/tools-header.tsx

```diff
@@ -29,10 +30,26 @@ export default function ToolsHeader({ header, filter, pagination, preferences }:
       {isHeaderString ? <span id={headingId}>{header}</span> : header}
       {hasTools && (
         <div ref={ref} className={clsx(styles.tools, isSmall && styles['tools-small'])}>
-          {filter && <div className={styles['tools-filtering']}>{filter}</div>}
+          {filter && (
+            <div
+              className={styles['tools-filtering']}
+              onClickCapture={() => setLastUserAction?.('filter')}
+              onKeyDownCapture={() => setLastUserAction?.('filter')}
+            >
+              {filter}
+            </div>
+          )}
           <div className={styles['tools-align-right']}>
-            {pagination && <div className={styles['tools-pagination']}>{pagination}</div>}
-            {preferences && <div className={styles['tools-preferences']}>{preferences}</div>}
+            {pagination && (
+              <div className={styles['tools-pagination']} onClickCapture={() => setLastUserAction?.('paginate')}>
```

Comment:
nit: Do we use verbs or nouns for actions? The "paginate" and "preferences" look inconsistent in that regard.

Reply:
Is this one emitted when the pagination is activated with keyboard?

Reply:
It is indeed a bit of a mix currently. I've updated it now to be all nouns.

React also emits the click event when a button is triggered with the keyboard, so yes, keyboard interaction is tracked correctly here as well.